dissect/target/plugins/apps/ssh/openssh.py

```diff
@@ -190,3 +194,36 @@ def parse_known_host(known_host_string: str) -> tuple[str, list, str, str, str]:
     comment = " ".join(parts[3:]) if len(parts) > 3 else ""
 
     return marker, hostnames.split(","), keytype, public_key, comment
+
+
+def calculate_fingerprints(public_key_decoded: bytes, ssh_keygen_format: bool = False) -> tuple[str, str, str]:
+    """Calculate the MD5, SHA1 and SHA256 digest of the given decoded public key.
+
+    Adheres as much as possible to the output provided by ssh-keygen when ``ssh_keygen_format``
+    parameter is set to ``True``. When set to ``False`` (default) hexdigests are calculated
+    instead for ``sha1``and ``sha256``.
+
+    Resources:
+        - https://en.wikipedia.org/wiki/Public_key_fingerprint
+        - https://man7.org/linux/man-pages/man1/ssh-keygen.1.html
+        - ``ssh-keygen -l -E <alg> -f key.pub``
+    """
+    if not public_key_decoded:
+        raise ValueError("No decoded public key provided")
+
+    if not isinstance(public_key_decoded, bytes):
+        raise ValueError("Provided public key should be bytes")
+
+    if public_key_decoded[0:3] != b"\x00\x00\x00":
+        raise ValueError("Provided value does not look like a public key")
+
+    fingerprint_md5 = md5(public_key_decoded).hexdigest()
+
+    if ssh_keygen_format:
+        fingerprint_sha1 = base64.b64encode(sha1(public_key_decoded).digest()).rstrip(b"=").decode("utf-8")
+        fingerprint_sha256 = base64.b64encode(sha256(public_key_decoded).digest()).rstrip(b"=").decode("utf-8")
+    else:
+        fingerprint_sha1 = sha1(public_key_decoded).hexdigest()
+        fingerprint_sha256 = sha256(public_key_decoded).hexdigest()
+
+    return fingerprint_md5, fingerprint_sha1, fingerprint_sha256
```

Comment:
```suggestion
    digest_md5 = md5(public_key_decoded).digest()
    digest_sha1 = sha1(public_key_decoded).digest()
    digest_sha256 = sha256(public_key_decoded).digest()

    if ssh_keygen_format:
        fingerprint_sha1 = base64.b64encode(digest_sha1).rstrip(b"=").decode()
        fingerprint_sha256 = base64.b64encode(digest_sha256).rstrip(b"=").decode()
    else:
        fingerprint_sha1 = digest_sha1.hex()
        fingerprint_sha256 = digest_sha256.hex()

    return digest_md5.hex(), fingerprint_sha1, fingerprint_sha256
```
Or something along those lines. A little bit cleaner.